src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/WebhookGoCardlessService.java

```diff
@@ -60,21 +66,48 @@
 
     public void handleEvents(List<GoCardlessEvent> events) {
         events.forEach(goCardlessService::storeEvent);
+        updateStatesForEvents(events);
+        var mapOfBooleanToListOfEvents = events.stream().collect(partitioningBy(event -> shouldBeHandled(event)));
+        handleValidEvents(mapOfBooleanToListOfEvents.get(true));
+        handleInvalidEvents(mapOfBooleanToListOfEvents.get(false));
```

Comment:
Essentially these are events we might send emails for and events we log errors for, right?